rero_ils/modules/documents/dojson/contrib/marc21tojson/utils.py

```diff
@@ -1716,35 +1683,78 @@ def do_work_access_point(marc21, key, value):
+            remove_trailing_punctuation(form_subdivision, ',.')
+            for form_subdivision in list(utils.force_list(value.get('k')))
+        ]))
+    # work_access_point.medium_of_performance_for_music
     if value.get('m'):
         work_access_point['medium_of_performance_for_music'] = list(
             utils.force_list(value.get('m')))
+    # work_access_point.arranged_statement_for_music
     if value.get('o'):
         work_access_point['arranged_statement_for_music'] = not_repetitive(
-            marc21.bib_id, marc21.bib_id, key, value, 'o')
+            bib_id, bib_id, key, value, 'o')
+    # work_access_point.key_for_music
     if value.get('r'):
         work_access_point['key_for_music'] = remove_trailing_punctuation(
-            not_repetitive(marc21.bib_id, marc21.bib_id, key, value, 'r'),
-            ',.'
-        )
-    if identifier := build_identifier(value):
-        agent['identifiedBy'] = identifier
+            not_repetitive(bib_id, bib_id, key, value, 'r'), ',.')
 
-    if not work_access_point.get('title'):
-        error_print('WARNING WORK ACCESS POINT:', marc21.bib_id,
-                    marc21.rero_id, 'no title')
-        return None
-    agent = work_access_point.get('agent', {})
-    if agent and not agent.get('preferred_name'):
-        work_access_point.pop('agent')
     return work_access_point or None
 
 
+def _do_work_access_point_creator(marc21, key, value):
+    """Create the structure for the work_access_point.creator field.
+
+    :param marc21: the marc record.
+    :param key: the MARC tag code and indicator.
+    :param value: the MARC tag content (subfields).
+    :return the work_access_point.creator structure as a dict
+    :rtype dict
+    """
+    tag = key[:3]
+    bib_id = marc21.bib_id
+    data = {}
+    if tag in ['100', '700', '800']:
+        data = {'type': EntityType.PERSON}
+        if value.get('a'):
+            data['preferred_name'] = remove_trailing_punctuation(
+                not_repetitive(bib_id, bib_id, key, value, 'a',)).rstrip('.')
+        if value.get('b'):
+            data['numeration'] = remove_trailing_punctuation(
+                not_repetitive(bib_id, bib_id, key, value, 'b'))
+        if date := not_repetitive(bib_id, bib_id, key, value, 'd'):
+            date_parts = [d.strip().rstrip('.') for d in date.split('-')]
+            if date_parts[0]:
+                data['date_of_birth'] = date_parts[0]
+            if date_parts[1]:
+                data['date_of_death'] = date_parts[1]
+        if value.get('c'):
+            data['qualifier'] = remove_trailing_punctuation(
+                not_repetitive(bib_id, bib_id, key, value, 'c')).rstrip('.')
+
+    # bf:Organisation
+    if tag == '710':
+        data = {
+            'type': EntityType.ORGANISATION,
+            'conference': False
+        }
+        if value.get('a'):
+            data['name'] = remove_trailing_punctuation(
```

Comment:
Should be `preferred_name`?

Reply:
@rerowep The directives were "No change on "work_access_point" except field `author` renamed to `creator`